docs/CMakeLists.txt

```diff
@@ -6,15 +6,29 @@ else()
   find_package(Sphinx QUIET)
 endif()
 
-if(SPHINX_FOUND)
-  message(STATUS "Sphinx Found, configuring.")
+find_package(Doxygen)
+
+if (SPHINX_FOUND AND DOXYGEN_FOUND)
+  message(STATUS "Sphinx and Doxygen found, configuring.")
+
   set(SPHINX_SOURCE ${CMAKE_SOURCE_DIR}/docs)
   set(SPHINX_BUILD ${CMAKE_CURRENT_BINARY_DIR})
   set(SPHINX_MAN ${CMAKE_CURRENT_BINARY_DIR}/man)
 
+  # We need to run doxygen before sphinx to update doxygen/tagfile.xml. This
+  # command does it. The output is a dummy one so the command always runs.
+  # doxygen has some level of caching so it doesn't take that long, and
+  # collecting a list of all source files would be cumbersome.
+  add_custom_command(OUTPUT dummy-doxygen
```

Comment:
I would not call this "dummy", I would simply call this what it is. You are building doxy. I'd prefer it be in /build tho, vs the root so we don't need another .gitignore.

Reply:
> I would not call this "dummy", I would simply call this what it is. You are building doxy.

The output is really a dummy one. CMake expects a file there, but I give something that never gets produced to force running `doxygen` every time. This is what the comment above tries to explain.

> I'd prefer it be in /build tho, vs the root so we don't need another .gitignore.

The output folder is set in the `Doxyfile` and cannot be overridden on the command line. I cannot easily change it because I need to run `doxygen` in the RTD builder, and installing all the dependencies there would be overkill. Maybe I can come up with some CMake magic to configure the `Doxyfile` on RTD...